src/Microsoft.DotNet.Build.Tasks/AddDependenciesToProjectJson.cs

```diff
@@ -111,14 +117,75 @@ public override bool Execute()
             // Update framework dependencies sections
             for (int i = 0; i < Frameworks.Length; i++)
             {
-                dependencies = GenerateDependencies(projectRoot, ExternalPackages, Frameworks[i]);
+                dependencies = GenerateDependencies(projectRoot, ExternalPackages, packageInformation, Frameworks[i]);
                 projectRoot = UpdateDependenciesProperty(projectRoot, dependencies, Frameworks[i]);
             }
             WriteProject(projectRoot, OutputProjectJson);
 
             return true;
         }
 
+        /// <summary>
+        /// Given a package name regex pattern, and an array of drop locations, create an array of objects
+        /// containing package information (name, version,, prerelease version)
+        /// </summary>
+        /// <param name="packagesDrops"></param>
+        /// <returns></returns>
+        private IEnumerable<ITaskItem> GatherPackageInformationFromDrops(string [] packagesDrops)
+        {
+            List<ITaskItem> packageNameItems = new List<ITaskItem>();
+
+            foreach (string packageDrop in packagesDrops)
+            {
+                if (!Directory.Exists(packageDrop))
+                {
+                    Log.LogWarning("PackageDrop does not exist - '{0}'", packageDrop);
+                    continue;
+                }
+                IEnumerable<ITaskItem> packages = Directory.GetFiles(packageDrop).Select(f => new TaskItem(Path.GetFileNameWithoutExtension(f)));
+
+                foreach (ITaskItem package in packages)
+                {
+                    packageNameItems.Add(CreatePackageItemFromString(package.ItemSpec));
+                }
+            }
+            return packageNameItems;
+        }
+
+        // A versions file is of the form https://github.com/dotnet/versions/blob/master/build-info/dotnet/corefx/master/Latest_Packages.txt
+        private IEnumerable<ITaskItem> GatherPackageInformationFromVersionsFile(string versionsFile)
+        {
+            List<ITaskItem> packageNameItems = new List<ITaskItem>();
+            if(!File.Exists(versionsFile))
+            {
+                Log.LogError("Specified versions file ({0}) does not exist.", versionsFile);
+            }
+            var lines = File.ReadAllLines(versionsFile);
+            foreach(string line in lines)
+            { 
+                if(!string.IsNullOrWhiteSpace(line))
+                {
+                    string packageVersion = line.Replace(' ', '.');
```

Comment:
Just split on " " and create an override of CreatePackageItemFromString that takes the ID and version seperately.  Then use NuGet to parse the version rather than trying to roll our own parsing.